### PR TITLE
Non-functional improvements of the Jail helpers

### PR DESCRIPTION
**Pull request by Ashod**
Non-functional changes. Non-disruptive to allow cherry-picking to release branches with minimal effort (no large code movements, for example).

Each commit is self contained and has more explanation in the commit-message.

- wsd: jail: rename jails to childroot to distinguish them
- wsd: jail: move LO_JAIL_SUBPATH to JailUtil
- wsd: jail: no need to pass loSubPath between processes
- wsd: jail: add Poco namespace where missing in LinkOrCopy
- wsd: jail: encapsulate LinkOrCopy into a class
- wsd: jail: unlink before copying files
- wsd: jail: simplify linkOrCopy member names

### CODE REVIEW

File: kit/Kit.cpp [hunk 1]
```diff
@@ -374,42 +365,39 @@ class LinkOrCopy
...
+        FileCount = 0;
+        StartTime = std::chrono::steady_clock::now();
 
-        if (nftw(source.c_str(), linkOrCopyFunction, 10, FTW_ACTIONRETVAL|FTW_PHYS) == -1)
+        if (nftw(source.c_str(), linkOrCopyFunction, 10, FTW_ACTIONRETVAL | FTW_PHYS) == -1)
         {
             LOG_SYS("linkOrCopy: nftw() failed for '" << source << '\'');
         }
 
-        if (linkOrCopyVerboseLogging)
```
Comment (mmeeks) [inline kit/Kit.cpp:384]: The code change -looks- as if we now un-conditionally LOG_INF in this case, add another ::now() call into that code-path etc.

File: kit/Kit.cpp [hunk 2]
```diff
@@ -300,12 +299,6 @@ class LinkOrCopy
             break;
         case FTW_D:
             {
-                struct stat st;
```
Comment (mmeeks) [inline kit/Kit.cpp:297]: That looks like a functional change to the casual observer; that's very unexpected in something that says it just addresses renames.

**Review by mmeeks — CHANGES_REQUESTED** [review]
I read the first very large chunk of change which I hoped would be a slam-dunk approval - and found an apparent functional change in a patch described as renaming things. Possibly it's the right change - but I'd like love to review that as a separate commit.

**Review by mmeeks — APPROVED** [review]
Sounds sensible; then again - we should not have permission to overwrite those files anyway - if we do, there is a problem so ...

**Review by mmeeks — CHANGES_REQUESTED** [review]
Perhaps I'm reading this backwards - but what is the benefit of class with purely static methods over just using nftw ? hmm?

File: kit/Kit.cpp (hunk 2 above)
Comment (Ashod) [reply kit/Kit.cpp:297]: We are given the file's stat. Why are we doing another stat call? No functional change, just using the existing stat results.

**Comment by mmeeks** [comment]
Hi Ash; soo .... this has been lingering for over a year - I wonder if we should (when there is time after the 21.11 release) - re-visit these if they're still relevant (?) or should we close this ?
Thanks ! =)

**Pull request closed without merging**